infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/plugins/DefaultClusterManager.java

```diff
@@ -182,199 +204,339 @@ public TopologyDescriptor readConfig() {
+                        sinkClusterIds.get(i), sinkNodeIds.get(i), sinkNodeIds.get(i));
+                nodes.add(nodeInfo);
             }
-            sinkClusters.add(sinkCluster);
+            ClusterDescriptor sinkCluster = new ClusterDescriptor(sinkClusterIds.get(i),
+                    Integer.parseInt(sinkCorfuPorts.get(i)), nodes);
+
+            sinkClustersToReplicationModel.put(sinkCluster,
+                    new HashSet<LogReplication.ReplicationModel>(){
+                        {
+                            add(LogReplication.ReplicationModel.FULL_TABLE);
+                        }});
+            allClusters.put(sinkCluster.getClusterId(), sinkCluster);
         }
 
-        return new TopologyDescriptor(0L, sourceClusters, sinkClusters, localEndpoint);
-    }
+        // create a backup Cluster
+        List<NodeDescriptor> nodes = Arrays.asList( new NodeDescriptor(
+                topology.getDefaultHost(),
+                topology.getBackupLogReplicationPort(),
+                DefaultClusterConfig.getBackupClusterIdx().get(0),
+                topology.getBackupNodesUuid().get(0),
+                topology.getBackupNodesUuid().get(0)
+        ));
+        ClusterDescriptor backupCluster = new ClusterDescriptor(DefaultClusterConfig.getBackupClusterIdx().get(0),
+                BACKUP_CORFU_PORT, nodes);
+        allClusters.put(backupCluster.getClusterId(), backupCluster);
 
-    private TopologyConfigurationMsg constructTopologyConfigMsg() {
-        TopologyDescriptor clusterTopologyDescriptor = readConfig();
-        return clusterTopologyDescriptor.convertToMessage();
+        return new TopologyDescriptor(0L, sinkClustersToReplicationModel, sourceClustersToReplicationModel,
+                allClusters, new HashSet<>(), localNodeId);
     }
 
 
     @Override
-    public void register(CorfuReplicationDiscoveryService corfuReplicationDiscoveryService) {
-        this.corfuReplicationDiscoveryService = corfuReplicationDiscoveryService;
+    public TopologyDescriptor queryTopologyConfig(boolean useCached){
+        try {
+            log.info("Wait until a topology is created");
+            waitForTopologyInit.await();
+        } catch (InterruptedException e) {
+            log.error("Interrupted");
+        }
+        log.debug("new cluster config msg " + topologyConfig);
+        return topologyConfig;
     }
 
-    @Override
-    public void setLocalEndpoint(String endpoint) {
-        this.localEndpoint = endpoint;
+    private void createSingleSourceSinkTopology() {
+        topologyConfig = generateDefaultValidTopology();
+
+        waitForTopologyInit.countDown();
     }
 
-    @Override
-    public TopologyConfigurationMsg queryTopologyConfig(boolean useCached) {
-        if (topologyConfig == null || !useCached) {
-            topologyConfig = constructTopologyConfigMsg();
+    private TopologyDescriptor generateDefaultValidTopology() {
```

Comment:
can you rename this method to something more specific?

Reply:
removed.